### DISCUSSION TITLE
Confusing about read and write quorum in Erasure Code Calculator !!

### DISCUSSION THREAD

**Discussion by kietheros** [minio/minio#19304] [Q&A]
The annotations in erasure-code-calculator:
`
If the Erasure Code Parity (M) is 1/2 the Stripe Size, read + write quorum is (Stripe Size - (M + 1)) drives.
`
I read from other docs, if K=M=N/2, read quorum is K = N/2, write quorum is **K+1 = N - M + 1** = N/2+1. I don't understand why read+write quorum is `(N - (M + 1))`. And why does this document combine read quorum and write quorum? If N/2 disks fail, cannot write to Minio, so it cannot be said that the system still works.
![Screenshot from 2024-03-20 23-51-09](https://github.com/minio/minio/assets/44565784/5c2760ed-ea1b-49ae-822a-b58f2fd30707)

**Comment by harshavardhana** [comment]
So reading is not essential to you? @kietheros, it says your tolerance is only 1 node, which is the WRITE quorum. 

Where does it say that you can tolerate n/2?

**Reply by kietheros** [reply]
@harshavardhana With the above MinIO settings (2 servers, 8 disks, K=M=N/2=4), I understand the write quorum is K+1=5. If just one server fails, we lose 4 disks, leaving only 4 disks remaining, less than write quorum so we cannot write to MinIO. I think server failure tolerance is 0. Please correct me if I am wrong !

**Reply by harshavardhana** [reply]
With 2 servers there is no tolerance for writes. You need a minimum of 3 servers to have a tolerance of 1.

**Reply by kietheros** [reply]
Thank you ! I understand that. I'm just confused about the description on the calculator website.
